precompiles/authorization/IICS20Authorization.sol

```diff
@@ -0,0 +1,70 @@
+    /// @param allocations An array of Allocation authorized with this grant.
+    event IBCTransferAuthorization(
+        address indexed grantee,
+        address indexed granter,
+        Allocation[] allocations
+    );
+
+    /// @dev Approves IBC transfer with a specific amount of tokens.
+    /// @param grantee The address for which the transfer authorization is granted.
+    /// @param allocations An array of Allocation for the authorization.
+    function approve(address grantee, Allocation[] calldata allocations) external returns (bool approved);
+
+    /// @dev Revokes IBC transfer authorization for a specific grantee.
+    /// @param grantee The address for which the transfer authorization will be revoked.
+    function revoke(address grantee) external returns (bool revoked);
+
+    /// @dev Increase the allowance of a given grantee by a specific amount of tokens for IBC transfer methods.
+    /// @param grantee The address of the contract that is allowed to spend the granter's tokens.
+    /// @param sourcePort The port on which the packet will be sent.
+    /// @param sourceChannel The channel by which the packet will be sent.
+    /// @param denom The denomination of the Coin to be transferred to the receiver
+    /// @param amount The increase in amount of tokens that can be spent.
+    /// @return approved Is true if the operation ran successfully.
+    function increaseAllowance(
+        address grantee,
+        string calldata sourcePort,
+        string calldata sourceChannel,
+        string calldata denom,
+        uint256 amount
+    ) external returns (bool approved);
+
+    /// @dev Decreases the allowance of a given grantee by a specific amount of tokens for for IBC transfer methods.
+    /// @param grantee The address of the contract that is allowed to spend the granter's tokens.
+    /// @param sourcePort The port on which the packet will be sent.
+    /// @param sourceChannel The channel by which the packet will be sent.
+    /// @param denom The denomination of the Coin to be transferred to the receiver.
+    /// @param amount The decrease in amount of tokens that can be spent.
+    /// @return approved Is true if the operation ran successfully.
+    function decreaseAllowance(
+        address grantee,
+        string calldata sourcePort,
+        string calldata sourceChannel,
+        string calldata denom,
+        uint256 amount
+    ) external returns (bool approved);
+
+    /// @dev Returns the remaining number of tokens that a grantee smart contract
+    /// will be allowed to spend on behalf of granter through
+    /// IBC transfers. This is an empty array by default.
+    /// @param grantee The address of the contract that is allowed to spend the granter's tokens.
+    /// @param granter The address of the account able to transfer the tokens.
+    /// @return allocations The remaining amounts allowed to spend for
+    /// corresponding source port and channel.
```

Comment:
The source port and channel are not included in the function, so this comment is a little confusing. Could be explained better IMO.
```suggestion
    /// @return allocations The remaining amounts allowed to be spent for
    /// corresponding source port and channel.
```